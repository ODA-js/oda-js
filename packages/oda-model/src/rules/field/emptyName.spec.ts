import Rule from './emptyName';

describe('emptyName', () => {
  let rule: Rule;
  beforeAll(() => {
    rule = new Rule();
  });

  it('throw', () => {

    const result = rule.validate({
      field: {
      },
    } as any);

    expect(result).toMatchSnapshot();

  });

  it('not throw', () => {
    const result = rule.validate({
      field: {
        name: 'species',
      },
    } as any);

    expect(result).toMatchSnapshot();

  });
});
