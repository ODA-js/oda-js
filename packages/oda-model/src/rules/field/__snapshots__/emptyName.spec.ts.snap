// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emptyName not throw 1`] = `Array []`;

exports[`emptyName throw 1`] = `
Array [
  Object {
    "message": "field must be named",
    "result": "error",
  },
]
`;
